java/src/jmri/jmrit/beantable/StringIOTableAction.java

```diff
@@ -0,0 +1,319 @@
+            ActionListener cancelListener = this::cancelPressed;
+            ActionListener rangeListener = this::canAddRange;
+            configureManagerComboBox(prefixBox, stringIOManager, StringIOManager.class);
+            userNameTextField.setName("userName"); // NOI18N
+            prefixBox.setName("prefixBox"); // NOI18N
+            addButton = new JButton(Bundle.getMessage("ButtonCreate"));
+            addButton.addActionListener(createListener);
+
+            if (hardwareAddressValidator==null){
+                hardwareAddressValidator = new SystemNameValidator(hardwareAddressTextField, java.util.Objects.requireNonNull(prefixBox.getSelectedItem(), "encountered null system selection"), true);
+            } else {
+                log.trace("on add, prefixBox.getSelected is {}", prefixBox.getSelectedItem());
+                hardwareAddressValidator.setManager(prefixBox.getSelectedItem());
+            }
+
+            // create panel
+            addFrame.add(new AddNewHardwareDevicePanel(hardwareAddressTextField, hardwareAddressValidator, userNameTextField, prefixBox,
+                    numberToAddSpinner, rangeCheckBox, addButton, cancelListener, rangeListener, statusBarLabel));
+            // tooltip for hardwareAddressTextField will be assigned next by canAddRange()
+            canAddRange(null);
+        }
+        hardwareAddressTextField.setName("sysName"); // for GUI test NOI18N
+        hardwareAddressTextField.setName("hwAddressTextField"); // for GUI test NOI18N
+        addButton.setName("createButton"); // for GUI test NOI18N
+        // reset statusBarLabel text
+        statusBarLabel.setText(Bundle.getMessage("HardwareAddStatusEnter"));
+        statusBarLabel.setForeground(Color.gray);
+        addFrame.setEscapeKeyClosesWindow(true);
+        addFrame.getRootPane().setDefaultButton(addButton);
+        addFrame.pack();
+        addFrame.setVisible(true);
+    }
+
+    void cancelPressed(ActionEvent e) {
+        removePrefixBoxListener(prefixBox);
+        addFrame.setVisible(false);
+        addFrame.dispose();
+        addFrame = null;
+    }
+
+    /**
+     * Respond to Create new item button pressed on Add StringIO pane.
+     *
+     * @param e the click event
+     */
+    void createPressed(ActionEvent e) {
+
+        int numberOfStringIOs = 1;
+
+        if (rangeCheckBox.isSelected()) {
+            numberOfStringIOs = (Integer) numberToAddSpinner.getValue();
+        }
+        if (numberOfStringIOs >= 65 // limited by JSpinnerModel to 100
+            && JmriJOptionPane.showConfirmDialog(addFrame,
+                Bundle.getMessage("WarnExcessBeans", Bundle.getMessage("StringIOs"), numberOfStringIOs),
+                Bundle.getMessage("WarningTitle"),
+                JmriJOptionPane.YES_NO_OPTION ) != JmriJOptionPane.YES_OPTION ) {
+            return;
+        }
+        String rName;
+        String stringIOPrefix = prefixBox.getSelectedItem().getSystemPrefix(); // Add "R" later
```

Comment:
`// Add "R" later`
What does `R` stands for? Copy/paste error?

Reply:
I don't know what this comment means.  it comes from the Reporter support, which I copied to create this.  I'll remove it from StringIO and leave it in Reporter.